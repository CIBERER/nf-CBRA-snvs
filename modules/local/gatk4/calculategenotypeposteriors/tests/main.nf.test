nextflow_process {

    name "Test Process GATK4_CALCULATEGENOTYPEPOSTERIORS"
    script "../main.nf"
    process "GATK4_CALCULATEGENOTYPEPOSTERIORS"

    tag "modules"
    tag "modules_nfcore"
    tag "gatk4"
    tag "gatk4/calculategenotypeposteriors"


    test("homo_sapiens - [vcf, tbi, [], ped]") {

        when {
            process {
                """
                input[0] = [
                    [id:"test"],
                    file('/mnt/tblab/yolanda/analysis_trios/merged.vcf.gz', checkIfExists: true),
                    file('/mnt/tblab/yolanda/analysis_trios/merged.vcf.gz.tbi', checkIfExists: true),
                    [],
                    file('/mnt/tblab/yolanda/analysis_trios/ped_file.ped', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
