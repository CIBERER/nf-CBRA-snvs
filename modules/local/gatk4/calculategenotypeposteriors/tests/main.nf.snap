{
    "homo_sapiens - [vcf, tbi, [], ped]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz:md5,2bfb7d98c9030213408ca46ac69f822f"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz.tbi:md5,5ea611aacd8d5eb2971ae98259e3690a"
                    ]
                ],
                "2": [
                    "versions.yml:md5,8b428765690c05516290de723f660472"
                ],
                "tbi": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz.tbi:md5,5ea611aacd8d5eb2971ae98259e3690a"
                    ]
                ],
                "vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz:md5,2bfb7d98c9030213408ca46ac69f822f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8b428765690c05516290de723f660472"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-17T10:15:07.043393553"
    }
}